tests/test_ast.py

```diff
@@ -162,3 +163,108 @@ def test_string_escaped():
     t = a.statements[0]
     assert_is_instance(t, ast.StringNode)
     assert_equal(t.string, 'this is a \\"string')
+
+
+def test_fun_call_empty():
+    a = parse("foo()")
+    assert_equal(1, len(a.statements))
+    c = a.statements[0]
+    assert_is_instance(c, ast.FunctionCallNode)
+    assert_equal("foo", c.fun.text)
+    assert_is_none(c.args)
+
+
+def test_fun_call_one():
+    a = parse("foo(1)")
+    assert_equal(1, len(a.statements))
+    c = a.statements[0]
+    assert_is_instance(c, ast.FunctionCallNode)
+    assert_equal("foo", c.fun.text)
+    assert_equal("1", c.args.text)
+
+
+def test_fun_call_many():
+    a = parse("foo(1, 2, 3)")
+    assert_equal(1, len(a.statements))
+    c = a.statements[0]
+    assert_is_instance(c, ast.FunctionCallNode)
+    assert_equal("foo", c.fun.text)
+    assert_equal("1", c.args.lhs.text)
+    assert_equal("2", c.args.rhs.lhs.text)
+    assert_equal("3", c.args.rhs.rhs.text)
+
+
+def test_return():
+    a = parse("return 1 + 2")
+    assert_equal(1, len(a.statements))
+    r = a.statements[0]
+    assert_is_instance(r, ast.StatementReturnNode)
+    assert_is_instance(r.rhs, ast.OperatorAddNode)
+    assert_equal("1", r.rhs.lhs.text)
+    assert_equal("2", r.rhs.rhs.text)
+
+
+def test_fun_def_void_empty():
+    a = parse("fun foo() endfun")
+    assert_equal(1, len(a.statements))
+    f = a.statements[0]
+    assert_is_instance(f, ast.StatementFunNode)
+    s = f.signature
+    assert_is_instance(s, ast.FunctionCallNode)
+    assert_equal("foo", s.fun.text)
+    assert_is_none(s.args)
+    assert_equal(0, len(f.children))
+
+
+def test_fun_def_void():
+    a = parse("""
+    fun foo(a: u8, b: u16)
+       return a + b
```

Comment:
this is a good place to test a bare `return` rather than `return a + b`.

i think the tests should be valid syntax unless we're specifically testing parse errors. and since this function doesn't have a return type, it's invalid to return an expression.